docs/porposals/diamond.md

```diff
@@ -0,0 +1,27 @@
+# Diamond workflow support.
+
+Diamond workflow allows for a pipeline to scale out the generation of a single bundle.
+
+The diamond workflow consists of the following steps.
+
+1. Before a diamond starts, Datamon SDK allows for the creation of an unique ID for a diamond.
+   1. This can be a KSUID.
+   2. The client is expected to use the KSUID at each step.
+   3. command: ```datamon bundle split initialize --repo <repo> --message```
+   
+2. Each node in a diamond generates a filelist that is stored in the versioned metadata folder.
+   1. Each filelist is given a client assigned ID. Example node1,node2,node3 or 1,2,3 or a UUID.
+   2. On retries the ID should be the same for a given node.
+   3. command: ```datamon bundle split upload --repo <repo> --split-id <id> --source <path>```
```

Comment:
NOTE: all kind of distributed commit management involves a "pre-commit" stage. The main takeaway from the most popular alg (2-phase-commit) is that the final commit requires all the pre-committed parts to be complete. In my opinion, this may be achieve with some kind of flag file in the staging metadata (e.g. `bundles/{bundleID}/precommit/{diamonID}/.ready`)

Reply:
I will add the backend layout I have in mind. Will add more details for failure handling.